Increase the request_timeout from default

rooftopcellist opened the pull request:
- As AWX gets overloaded with jobs, the response time of HTTP requests to the api/v2/jobs endpoint
- The result is that AnsibleJob objects launcht he job, then keep checking for the status of the job that was launch. The GET request times out, and it launches a new job, which complicates the problem further.
- This behavior is seen when createing 100+ AnsibleJob objects at once
- We should keep this HTTP request_timeout abnormally high until the resource operator grows a better way to handle this (smart checks for timeout errors, etc.)

dsavineau reviewed (changes requested):
We need to update the minimal awx.awx ansible collection version since the request_timeout parameter has been added in awx 22.7.0

Also would it make sense to make it a global parameter at CRD level so one hitting the hardcoded 600 seconds timeout could increase it in the CR ?

https://github.com/ansible/awx/commit/abc56305cc7ab0feb38c5b8c786f015ee7ed8b82

rooftopcellist commented:
@dsavineau That is a good idea.  I just added the ability to configure request_timeout on the job and workflow CR's.  I also added a better description for what Job TTL is while there.  

I bumped the minimum version of awx.awx collection to 22.7.0, which is the version that option was introduced in.

dsavineau commented on config/crd/bases/tower.ansible.com_ansiblejobs.yaml:
```diff
@@ -40,6 +40,8 @@ spec:
               workflow_template_name:
                 type: string
+              request_timeout:
+                type: integer
```
Shouldn't we add a description to this too?

dsavineau commented on config/crd/bases/tower.ansible.com_ansibleworkflows.yaml:
```diff
@@ -38,6 +38,8 @@ spec:
               workflow_template_name:
                 type: string
+              request_timeout:
+                type: integer
```
ditto

dsavineau commented on config/manifests/bases/awx-resource-operator.clusterserviceversion.yaml:
```diff
@@ -61,6 +62,11 @@ spec:
+        description: Time in seconds to wait for a response from the AWX API
+        path: request_timeout
+        x-descriptors:
+        - urn:alm:descriptor:com.tectonic.ui:text
```
maybe add `urn:alm:descriptor:com.tectonic.ui:advanced` too ?
What do you think ?

rooftopcellist replied:
agreed

dsavineau commented on config/manifests/bases/awx-resource-operator.clusterserviceversion.yaml:
```diff
@@ -198,10 +204,16 @@ spec:
+        description: Time in seconds to wait for a response from the AWX API
+        path: request_timeout
+        x-descriptors:
+        - urn:alm:descriptor:com.tectonic.ui:text
```
ditto

dsavineau commented on roles/job_runner/defaults/main.yml:
```diff
@@ -0,0 +1,4 @@
+---
+
+# Time before request to AWX api/v2/jobs/{id}/stdout/ times out (in seconds)
+request_timeout: 600
```
I think we need this default value in other roles too right ? the current parameter is added for ansiblejob and ansibleworkflow but those are using job and workflow roles rather tha job_runner if I'm not mistaken

rooftopcellist replied:
@dsavineau The ansiblejob and ansibleworkflow roles actually both call the job_runner role to create a k8s pod to run the playbook in.

We only have this issue with the job_wait tasks that check if the job is complete or not.  There are only 3 references, and they are all in the job_runner role:
* https://github.com/search?q=repo%3Aansible%2Fawx-resource-operator%20job_wait&type=code
